src/main/java/net/dries007/tfc/objects/entity/animal/EntityAnimalTFC.java

```diff
@@ -412,4 +417,39 @@ public void onLivingUpdate()
             }
         }
     }
+
+    public static void addCommonLivestockAI(EntityAnimal entity)
+    {
+        entity.tasks.addTask(2, new EntityAIMate(entity, 1.0D));
+
+        for (ItemStack is : OreDictionary.getOres("grain"))
+        {
+            Item item = is.getItem();
+            entity.tasks.addTask(3, new EntityAITempt(entity, 1.1D, item, false));
+        }
+        entity.tasks.addTask(6, new EntityAIEatGrass(entity));
+    }
+
+    public static void addCommonPreyAI(EntityAnimal entity, double speedMult)
+    {
+        double farSpeed = .5D * speedMult;
+        double nearSpeed = 1.1D * speedMult;
+
+        entity.tasks.addTask(0, new EntityAISwimming(entity));
+        entity.tasks.addTask(1, new EntityAIPanic(entity, 1.4D * speedMult));
+        //space for livestock AIMate and AITempt
+        entity.tasks.addTask(4, new EntityAIAvoidEntity<>(entity, EntityPlayer.class, 12.0F, farSpeed, nearSpeed));
```

Comment:
One minor nitpick: Did you test what this causes to familiarized animals? Because i think this will make it run from players regardless of familiarity stat.

Suggestion: Create an `EntityAIAvoidPlayer` extension from vanilla's with the player class already set and add the familiarity check. Anything non-zero sholdn't run from you since it would means it's not scared of you anymore.

Reply:
Done. Also applied it to Wolves!